### PR TITLE
[MXNET-267] Reduce ndarray size in test which produces a huge memory spike which …

### PR DESCRIPTION
**Pull request by larroy**
…crashes on some platforms

## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by piiswrong** [comment]
@sxjscience I think this was to catch an issue?

**Comment by eric-haibin-lin** [comment]
The big shape is intended for the test because the operator fails to handle such shape previously. Maybe we can move unit tests with large memory consumption to nightly tests?

**Comment by larroy** [comment]
What was the reason of fail in such a big shape? Anyways, the overal situation with the tests on devices requires quite some attention, is not a single test problem. Was just a low hanging fruit.

**Comment by sxjscience** [comment]
I think it's originally added to test the correctness of sorting a very large ndarray.

**Comment by marcoabreu** [comment]
Do we have the issue documented somewhere?

**Comment by sxjscience** [comment]
https://github.com/apache/incubator-mxnet/pull/8398/files

File: tests/python/unittest/test_ndarray.py [hunk 1]
```diff
@@ -590,9 +590,8 @@ def gt_topk(dat, axis, ret_typ, k, is_ascend):
     gt = gt_topk(a_npy, axis=None, ret_typ="indices", k=5*5*5*5, is_ascend=False)
     assert_almost_equal(nd_ret_argsort, gt)
 
-    # test topk with a big shape
-    a = mx.nd.arange(0, 54686454, step=1, repeat=1)
-    assert_almost_equal(a.topk(k=54686454).asnumpy(), a.asnumpy()[::-1])
+    a = mx.nd.arange(0, 1024, step=1, repeat=1)
+    assert_almost_equal(a.topk(k=1024).asnumpy(), a.asnumpy()[::-1])
```
Comment (ZiyueHuang) [inline tests/python/unittest/test_ndarray.py]: `k = 1024` cannot test against the failure in https://github.com/apache/incubator-mxnet/issues/8303. What platforms does this test crash?

By the way, here is another test for big shape, https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_optimizer.py#L237

Comment (larroy) [reply tests/python/unittest/test_ndarray.py]: Raspberry PI, has only 1G of memory + swap. We are doing something wrong if we have to write a test with such a big allocation, if the type is wrong as in the issue that we have linked, this should be mocked, or tested in a smarted way. Not just allocating huge arrays willy-nilly.

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

Thanks!

File: tests/python/unittest/test_ndarray.py (hunk 1 above)
Comment (eric-haibin-lin) [reply tests/python/unittest/test_ndarray.py]: Agree that it's a bit heavy for an unit test. Shall we move it to the nightly test suite?

**Pull request closed without merging**